include/aws/common/byte_buf.h

```diff
@@ -103,47 +110,21 @@ AWS_COMMON_API int aws_byte_buf_cat(struct aws_byte_buf *dest, size_t number_of_
 }
 #endif
 
-static inline int aws_byte_buf_init(struct aws_allocator * allocator, struct aws_byte_buf * buf, size_t len) {
-    buf->buffer = (uint8_t*)aws_mem_acquire(allocator, len);
-    if (!buf->buffer) return aws_raise_error(AWS_ERROR_OOM);
-    buf->len = 0;
-    buf->capacity = len;
-    buf->allocator = allocator;
-    return AWS_OP_SUCCESS;
-}
-
-static inline void aws_byte_buf_clean_up(struct aws_byte_buf * buf) {
-    if (buf->allocator && buf->buffer) aws_mem_release(buf->allocator, (void *)buf->buffer);
-    buf->allocator = NULL;
-    buf->buffer = NULL;
-    buf->len = 0;
-    buf->capacity = 0;
-}
-
 /**
- * For creating a byte buffer from a constant string. (With no nulls, else strlen will not work.)
+ * For creating a byte buffer from a null-terminated string literal.
  */
-static inline struct aws_byte_buf aws_byte_buf_from_literal(const char *literal) {
-    struct aws_byte_buf buf;
-    buf.buffer = (uint8_t *)literal;
-    buf.len = strlen(literal);
-    buf.capacity = buf.len;
-    buf.allocator = NULL;
-    return buf;
-}
-
-static inline struct aws_byte_buf aws_byte_buf_from_c_str(const char *c_str, size_t len) {
+static inline struct aws_byte_buf aws_byte_buf_from_c_str(const char * c_str) {
     struct aws_byte_buf buf;
     buf.buffer = (uint8_t *)c_str;
-    buf.len = len;
-    buf.capacity = len;
+    buf.len = strlen(c_str);
+    buf.capacity = buf.len;
     buf.allocator = NULL;
     return buf;
 }
 
-static inline struct aws_byte_buf aws_byte_buf_from_array(const uint8_t *c_str, size_t len) {
+static inline struct aws_byte_buf aws_byte_buf_from_array(const void * bytes, size_t len) {
```

Comment:
why are we making this void * now, uint8_t * seems appropriate given 'byte_buf'. I also like forcing users to do an explicit cast from const char * to uint8_t * as a "I agree that I want you to do thiat and I approve" mechanism.

Reply:
I don't feel strongly about that either way, but I do feel strongly that we shouldn't have two separate functions for the (uint8_t *) case and the (char *) case. I was following @bdonlan's opinion that we should make this a void * to make it more easily usable no matter whether the original array was signed or unsigned. Your opinion is the opposite. While I do find having to do the explicit casts here a bit annoying, I agree that uint8_t * is more descriptive, so assuming mine is the tiebreaking vote, I will change it back to that, but I will not include a separate function for (char *) arrays.

Reply:
I agree about not having both char * and uint8_t *